site/src/pages/components/switch.explainer.mdx

```diff
@@ -0,0 +1,574 @@
+| `focused`   | `true`/`false`    | `false`       |                                                                                                                       |                                                                                                             |
+
+### Accessibility
+
+The control can be in a checked or unchecked state. [^6]
+
+The `<switch>` element can be assigned the default role of switch as described in the [Core Accessibility API Mappings](https://w3c.github.io/core-aam/#role-map-switch). [^7]
+
+Content for thumb and either track side can be defined by the author.
+
+#### Keyboard Navigation and Focus
+
+{/* - _Arrow vs tabbing key behavior_ */}
+{/* - _Modifier key effects (e.g. shift, ctrl, meta)_ */}
+{/* - _Does this component use focus delegation?_ */}
+
+| Key              | Description                            |
+| ---------------- | -------------------------------------- |
+| Space            | Should toggle the switch when focused. |
+| Enter (optional) | Should toggle the switch when focused. |
+
+#### Form Input
+
+{/* _Does this component integrate with form submission, form validation, etc.? Can integration be accomplish with standard APIs or are special work-arounds required?_ */}
+
+The switch is also a form element, acting as a mix between a button and a checkbox - a stateful button.
+
+##### Type button
+
+A JavaScript event listener is used to trigger the immediate action assigned to the switch. This can include calling a REST API or toggling a class on an element (e.g. dark mode).
+
+##### Type submit
+
+<div class="question">Feedback on this feature would be welcome.</div>
+
+For a more traditional, non-JavaScript approach, it's necessary to receive the state of the switch on the server. In order to facilitate the directness of the action, a switch can be of type "submit". Like a submit button, toggling the switch will immediately submit the associated form to the server. This enables creating a HTML/CSS only settings page while staying true to user expectations of how the component should behave.
+
+#### Use with Assistive Technology
+
+{/* _Are there any details about shadow dom, focus delegation, or aria attributes that need special attention?_ */}
+
+The switch element can be associated with the existing `role="switch"`.
+
+### Globalization
+
+<div class="question">
+  Is thumb in the "on" position on the left in a `rtl` directional environment?
+</div>
+
+### Security
+
+{/* _Are there any security implications surrounding the component?_ */}
+
+No security implications are expected.
+
+## Related issues
+
+- [switch: What customizability for animations should be supported?](https://github.com/openui/open-ui/issues/816)
+- [[switch]: Naming of the component and it's parts](https://github.com/openui/open-ui/issues/804)
+
+## Resources
+
+{/* _Any related resource links such as web standards, discussion threads, diagrams, etc._ */}
```

Comment:
there should be links to thinks like the ARIA Authoring practices switch component page -https://www.w3.org/WAI/ARIA/apg/patterns/switch

arguably the switch definition in the aria spec.

there are a number of a11y-related switch articles / demos that exist as well which could be linked to, but i think just linking to the ARIA resources would probably be good enough

Reply:
Added the links for the APG switch as well as the aria spec for `role="switch"`.